### PR TITLE
Use `Ember.assign()` if possible

### PR DESCRIPTION
**Pull request by Cryrivers**
As `Ember.merge` is deprecated since `Ember 2.5.0`

### CODE REVIEW

File: addon/ajax-request.js [hunk 1]
```diff
@@ -200,8 +200,8 @@ export default class AjaxRequest {
    */
   _getFullHeadersHash(headers) {
     const classHeaders = get(this, 'headers') || {};
-    const _headers = merge({}, classHeaders);
-    return merge(_headers, headers);
+    const _headers = assign({}, classHeaders);
```
Comment (rwjblue) [inline addon/ajax-request.js]: `assign` allows more than two arguments so we should be able to do this in a single invocation:

```js
return assign({}, classHeaders, headers);
```

Comment (rwjblue) [inline addon/ajax-request.js]: Ahh, but we can't do that until we can require `v2.5.0`, :crying_cat_face:

**Comment by rwjblue** [comment]
This LGTM

File: addon/ajax-request.js (hunk 1 above)
Comment (Cryrivers) [inline addon/ajax-request.js]: @rwjblue yep, `assign` could be `Ember.merge`. :trollface:

**Comment by alexlafroscia** [comment]
Hmm, @cryrivers CI doesn't seem very happy with the change, but it looks good to me once that is settled.

**Comment by Cryrivers** [comment]
@alexlafroscia not sure why those 5 test cases failed :cry:

**Comment by orf** [comment]
Looks like the deprecation has been removed: https://github.com/emberjs/ember.js/pull/13344

**Comment by alexlafroscia** [comment]
Yeah, in light of that fact, I'm going to close this.  Thanks for the PR though, @Cryrivers!

**Pull request closed without merging**